### PR TITLE
Added renaming to Pokemon.java

### PR DESCRIPTION
**Pull request by nateschickler0**
renamePokemon(String nickname) renames a pokemon to a given string.

### CODE REVIEW

File: src/main/java/com/pokegoapi/api/pokemon/Pokemon.java [hunk 1]
```diff
@@ -55,14 +57,31 @@ public Result transferPokemon() throws LoginFailedException, RemoteServerExcepti
...
+
+		ServerRequest serverRequest = new ServerRequest(RequestType.NICKNAME_POKEMON, reqMsg);
+		pgo.getRequestHandler().request(serverRequest);
+		pgo.getRequestHandler().sendServerRequests();
+
+		NicknamePokemonResponse response;
+		try {
+			response = NicknamePokemonResponse.parseFrom(serverRequest.getData());
+		} catch (InvalidProtocolBufferException e) {
+			return null;
```
Comment (jabbink) [inline src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: On all other places `throw new RemoteServerException(e);` is used so the error can be handled by the caller.

File: .gitignore [hunk 2]
```diff
@@ -154,3 +154,4 @@ fabric.properties
 
 
 
+/src/resources/.gitignore
```
Comment (jabbink) [inline .gitignore]: ?

Comment (nateschickler0) [inline .gitignore]: Mistakenly added this, will remove it.

File: src/main/java/com/pokegoapi/api/pokemon/Pokemon.java (hunk 1 above)
Comment (nateschickler0) [inline src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: The method throws RemoteServerException.
... `throws LoginFailedException, RemoteServerException {`

Comment (Grover-c13) [inline src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: the return null in the catch state should throw a RemoteServerException

Comment (nateschickler0) [inline src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: Got it. Adding now.

**Comment by jabbink** [comment]
:+1:

**Pull request merged**